lib/components/base-components/NormalComponent/NormalComponent_doInitialSilkscreenOverlapAdjustment.ts

```diff
@@ -0,0 +1,353 @@
+    const allComponents = db.pcb_component.list()
+    for (const pcbComp of allComponents) {
+      if (
+        pcbComp.pcb_component_id !== component.pcb_component_id &&
+        isWithinRadius(componentCenter, pcbComp.center, searchRadius)
+      ) {
+        collectSilkscreenElementsFromComponent(
+          pcbComp.pcb_component_id,
+          db,
+          elements,
+        )
+      }
+    }
+  }
+
+  return elements
+}
+
+/**
+ * Collect all silkscreen elements for a specific component
+ */
+function collectSilkscreenElementsFromComponent(
+  pcbComponentId: string,
+  db: any,
+  elements: SilkscreenElement[],
+): void {
+  // Collect silkscreen paths
+  const paths = db.pcb_silkscreen_path
+    .list()
+    .filter((p: any) => p.pcb_component_id === pcbComponentId)
+  for (const path of paths) {
+    elements.push({
+      bounds: getSilkscreenPathBounds(path),
+      layer: path.layer,
+      type: "path",
+      pcb_component_id: pcbComponentId,
+      id: path.pcb_silkscreen_path_id,
+    })
+  }
+
+  // Collect silkscreen rects
+  const rects = db.pcb_silkscreen_rect
+    .list()
+    .filter((r: any) => r.pcb_component_id === pcbComponentId)
+  for (const rect of rects) {
+    elements.push({
+      bounds: getSilkscreenRectBounds(rect),
+      layer: rect.layer,
+      type: "rect",
+      pcb_component_id: pcbComponentId,
+      id: rect.pcb_silkscreen_rect_id,
+    })
+  }
+
+  // Collect silkscreen circles
+  const circles = db.pcb_silkscreen_circle
+    .list()
+    .filter((c: any) => c.pcb_component_id === pcbComponentId)
+  for (const circle of circles) {
+    elements.push({
+      bounds: getSilkscreenCircleBounds(circle),
+      layer: circle.layer,
+      type: "circle",
+      pcb_component_id: pcbComponentId,
+      id: circle.pcb_silkscreen_circle_id,
+    })
+  }
+
+  // Collect silkscreen texts
+  const texts = db.pcb_silkscreen_text
+    .list()
+    .filter((t: any) => t.pcb_component_id === pcbComponentId)
+  for (const text of texts) {
+    elements.push({
+      bounds: getSilkscreenTextBounds(text),
+      layer: text.layer,
+      type: "text",
+      pcb_component_id: pcbComponentId,
+      id: text.pcb_silkscreen_text_id,
+    })
+  }
+}
+
+/**
+ * Get component center with caching to avoid repeated DB lookups
+ */
+function getComponentCenterFromCache(
+  component: NormalComponent<any>,
+  db: any,
+): { x: number; y: number } | null {
+  if (!component.pcb_component_id) return null
+
+  // Use cached center if available from component's parsed props
+  if (
+    component._parsedProps?.pcbX !== undefined &&
+    component._parsedProps?.pcbY !== undefined
+  ) {
+    return {
+      x: component._parsedProps.pcbX,
+      y: component._parsedProps.pcbY,
+    }
+  }
+
+  // Fallback to DB lookup (single query per component)
+  const pcbComponent = db.pcb_component.get(component.pcb_component_id)
+  return pcbComponent?.center || null
+}
```

Comment:
remove. This is a totally invalid way to get a component center. Component's have a function for getting the bounds DO NOT USE DB!